core/amber/src/main/scala/edu/uci/ics/backenderror/Error.scala

```diff
@@ -0,0 +1,13 @@
+package edu.uci.ics.backenderror
+
+/**
+  * @param errorName a descriptive name of the error
+  * @param errorAdditionalDetails some details about the error (eg: exception's stacktrace)
+  * @param errorType
+  */
+case class Error(errorName: String, errorType: ErrorType.Value, errorSource:ErrorSource.Value, errorAdditionalDetails: Map[String, String]) {
```

Comment:
Why do we have to define such an `Error` interface? 
I thought the proper way is to use HTTP/WS Response's code and message to pass those kinds of information to the frontend. Or are you wrapping the `Error` inside the Response?

Also wouldn't a backtrace be useful?

Reply:
yes. the error is passed inside the response. `errorAdditionalDetails` is a map which can be used by the developer to put all information like stacktrace.

Reply:
yes. the error is passed inside the response. Check WorkflowWebsocketResource.scala below. errorAdditionalDetails is a map which can be used by the developer to put all information like stacktrace.

Reply:
When will we create an Error object? I don't see any place we create Error instances. Will it be created automatically when we throw an exception on the backend?

Reply:
@shengquan-ni my current plan is that whenever we catch an exception that we can't ignore, we log an error using this API. Also, same goes for all cases when the engine code reaches a place which it shouldn't have.